### PR TITLE
Deal with both unified and componentised deploys

### PR DESCRIPTION
**Pull request by sihil**
## What does this change?
Adds logic to the deploy radiator to deal with both the existing componentised and the new unified deployments.

## What is the value of this and can you measure success?
Since switching to the unified deploy in CODE the radiator has been misleading. This resolves that by looking at the unified deploy if it is more recent than the individual components. It should also (in theory) correctly work when we start using the unified deploy in PROD.

## Does this affect other platforms - Amp, Apps, etc?
No.

## Screenshots
From this:
screen shot 2016-12-06 at 13 17 27

To this:
screen shot 2016-12-06 at 13 20 34

### CODE REVIEW

File: static/src/deploys-radiator/app/main.js [hunk 1]
```diff
@@ -34,14 +34,39 @@ const run = () => {
...
-    const filterWhitelisted = (deploys) => deploys
+    const unifiedWhitelist = List([
+        'all'
+    ]);
+    const filterServerWhitelisted = (deploys) => deploys
         .filter(deploy => serverWhitelist.contains(deploy.projectName))
         .toList();
+    const filterUnifiedWhitelisted = (deploys) => deploys
+        .filter(deploy => unifiedWhitelist.contains(deploy.projectName))
+        .toList();
```
Comment (jfsoul) [inline static/src/deploys-radiator/app/main.js]: Any reason not to combine these? (same goes for `serverTime` below) e.g.    

    const filterWhiteListed = (whitelist) => (deploys) => deploys
        .filter(...)
        .toList()

    const filterServerWhiteListed = filterWhiteListed(serverWhiteList)

File: static/src/deploys-radiator/app/main.js [hunk 2]
```diff
@@ -34,14 +34,39 @@ const run = () => {
...
+    const mostRecentOf = (serverDeploys, unifiedDeploys) => {
+        if (serverDeploys.isEmpty()) return unifiedDeploys;
+        if (unifiedDeploys.isEmpty()) return serverDeploys;
+        const serverTime = serverDeploys.sort((a, b) => a.time.getTime() - b.time.getTime()).last().time.getTime();
+        const unifiedTime = unifiedDeploys.sort((a, b) => a.time.getTime() - b.time.getTime()).last().time.getTime();
+        if (serverTime > unifiedTime)
```
Comment (jfsoul) [inline static/src/deploys-radiator/app/main.js]: ternary might be neater

File: static/src/deploys-radiator/app/main.js [hunk 3]
```diff
@@ -34,14 +34,39 @@ const run = () => {
...
         .filter(deploy => serverWhitelist.contains(deploy.projectName))
         .toList();
+    const filterUnifiedWhitelisted = (deploys) => deploys
+        .filter(deploy => unifiedWhitelist.contains(deploy.projectName))
+        .toList();
+    const mostRecentOf = (serverDeploys, unifiedDeploys) => {
+        if (serverDeploys.isEmpty()) return unifiedDeploys;
+        if (unifiedDeploys.isEmpty()) return serverDeploys;
+        const serverTime = serverDeploys.sort((a, b) => a.time.getTime() - b.time.getTime()).last().time.getTime();
+        const unifiedTime = unifiedDeploys.sort((a, b) => a.time.getTime() - b.time.getTime()).last().time.getTime();
```
Comment (jfsoul) [inline static/src/deploys-radiator/app/main.js]: Is this logic already in `getMostRecentDeploys` in `model-helpers.js`?

**Review by jfsoul — CHANGES_REQUESTED** [review]
Would be good to re-use `getMostRecentDeploys` if possible. Otherwise minor style comments that don't necessarily need addressing.

File: static/src/deploys-radiator/app/main.js (hunk 3 above)
Comment (sihil) [inline static/src/deploys-radiator/app/main.js]: Not quite, but I've tried using it and it seems to work as expected so the difference is possibly not important.

File: static/src/deploys-radiator/app/main.js (hunk 2 above)
Comment (sihil) [inline static/src/deploys-radiator/app/main.js]: Too much time in scala :)

**Review by jfsoul — APPROVED** [review]
Thanks 👍

**Pull request merged**